### PR TITLE
Add extentsion app by reusing database connectivity functionality

### PR DESCRIPTION
**Pull request by nathanlcarlson**
This uses the same script now to do two things, serve as the mapapp
and extapp for the myproxy-logon. Use the --mapapp and the --extapp
flags to specify which string to build and output.

### CODE REVIEW

File: idp_node/mapapp/retriever.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    return psycopg2.connect(connection_params)
+
+def build_grouprole(args, config_parser, esgf_home):
+    ''' Build the group and role string for a specified username '''
+    logging.info("Fetching groups and roles for %s", args.username)
+
+    section = "installer.properties"
+    esgf_host = config_parser.get(section, "esgf.host")
+    gateway = "https://{}/esgf-idp/openid/%".format(esgf_host)
```
Comment (William-Hill) [inline idp_node/mapapp/retriever.py]: Is that trailing percent sign supposed to be standalone or is this a typo?

File: idp_node/mapapp/retriever.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        grouprole_elements.append("group_" + group + "_role_" + role)
+    return ";".join(grouprole_elements)
+
+def get_openid(args, config_parser, esgf_home):
+    ''' Retrieve the OpenID for a specified username '''
+    logging.info("Fetching OpenID for %s", args.username)
+
+    section = "installer.properties"
+    esgf_host = config_parser.get(section, "esgf.host")
+    gateway = "https://{}/esgf-idp/openid/%".format(esgf_host)
```
Comment (William-Hill) [inline idp_node/mapapp/retriever.py]: Same question as above

File: idp_node/mapapp/retriever.py (hunk 1 above)
Comment (nathanlcarlson) [reply idp_node/mapapp/retriever.py]: It is the "wildcard" character for the LIKE clause in the SQL statements.

File: idp_node/mapapp/retriever.py (hunk 2 above)
Comment (nathanlcarlson) [reply idp_node/mapapp/retriever.py]: It is the "wildcard" character for the LIKE clause in the SQL statements.

**Pull request merged**